src/main/java/com/googlecode/lanterna/gui2/MultiWindowTextGUI.java

```diff
@@ -471,11 +474,11 @@ public synchronized WindowBasedTextGUI moveToTop(Window window) {
      * @return Itself
      */
     public synchronized WindowBasedTextGUI cycleActiveWindow(boolean reverse) {
-        if(windows.isEmpty() || windows.size() == 1 || activeWindow.getHints().contains(Window.Hint.MODAL)) {
+        if(windows.isEmpty() || windows.size() == 1 || (activeWindow != null && activeWindow.getHints().contains(Window.Hint.MODAL))) {
             return this;
         }
         Window originalActiveWindow = activeWindow;
-        Window nextWindow = getNextWindow(reverse, originalActiveWindow);
+        Window nextWindow = activeWindow == null  ? windows.get(0) : getNextWindow(reverse, activeWindow);
         while(nextWindow.getHints().contains(Window.Hint.NO_FOCUS)) {
             nextWindow = getNextWindow(reverse, nextWindow);
             if(nextWindow == originalActiveWindow) {
```

Comment:
I suspect from just looking at it, that cycling away from <null>, if all Windows are not focus-able, might result in an infinite loop, because getNextWindow() only cycles among the (not null) windows, but originalActiveWindow is null here....